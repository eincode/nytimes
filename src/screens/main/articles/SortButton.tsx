import React from "react"
import {
  TouchableOpacity,
  StyleSheet,
  Image,
  TouchableOpacityProps,
  StyleProp,
  ViewStyle,
} from "react-native"

import IC_SORT from "../../../../assets/ic_sort.png"
import metrics from "../../../config/metrics"

interface IProps extends TouchableOpacityProps {
  style: StyleProp<ViewStyle>
}

export default (props: IProps) => (
  <TouchableOpacity
    style={[styles.container, props.style]}
    onPress={props.onPress}
  >
    <Image source={IC_SORT} style={styles.icon} />
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    width: 50,
    height: 50,
    borderRadius: 25,
    backgroundColor: metrics.PRIMARY_COLOR,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowRadius: 5,
    shadowOpacity: 1,
    elevation: 10,
    alignItems: "center",
    justifyContent: "center",
  },

  icon: {
    width: 30,
    height: 30,
  },
})
