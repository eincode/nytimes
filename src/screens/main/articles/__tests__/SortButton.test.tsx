import React from "react"
import renderer from "react-test-renderer"

import SortButton from "../SortButton"

it("Sort button component renders correctly", () => {
  const button = renderer.create(<SortButton />).toJSON()
  expect(button).toMatchSnapshot()
})
