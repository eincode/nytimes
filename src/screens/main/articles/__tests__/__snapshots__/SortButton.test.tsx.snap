// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort button component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(43, 152, 240)",
      "borderRadius": 25,
      "elevation": 10,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 5,
      "width": 50,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/ic_sort.png",
      }
    }
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  />
</View>
`;
